src/widgets/sliderwidget.cpp

```diff
@@ -301,6 +301,8 @@ void Amarok::VolumeSlider::paintEvent(QPaintEvent*) {
   const int padding = 7;
   const int offset = int(double((width() - 2 * padding) * value()) / maximum());
 
+  m_pixmapInset = volumePixmapDraw();
```

Comment:
yes, the function call inside the paint event is only needed when the theme changes. as you mentioned, I thought that it wont be that expensive to put it inside the paint event. However, it cannot be omitted as clementine would need a restart for the theme to take effect otherwise. if the foreground color get changed manually from clementine (appearance settings), it wont even detect the change. 
I modified the code such that the function get called only in case of a theme change.